src/gui/gui_effect_tooltips.cpp

```diff
@@ -6243,7 +6243,7 @@ uint32_t internal_make_effect_description(EFFECT_DISPLAY_PARAMS) {
 
 void effect_description(sys::state& state, text::layout_base& layout, dcon::effect_key k, int32_t primary_slot, int32_t this_slot,
 		int32_t from_slot, uint32_t r_lo, uint32_t r_hi) {
-	effect_tooltip::internal_make_effect_description(state, state.effect_data.data() + k.index(), layout, primary_slot, this_slot,
+	effect_tooltip::internal_make_effect_description(state, state.effect_data.data() + state.effect_data_indices[k.index()], layout, primary_slot, this_slot,
```

Comment:
For example, if this is called with the invalid handle, k.index() will equal -1, which is obviously not a good idea